src/WebJobs.Extensions.DurableTask/Listener/TaskOrchestrationShim.cs

```diff
@@ -75,7 +82,7 @@ public void SetFunctionInvocationCallback(Func<Task> callback)
                 Task invokeTask = this.functionInvocationCallback();
                 if (invokeTask is Task<object> resultTask)
                 {
-                    returnValue = await resultTask;
+                    returnValue = orchestratorInfo.IsOutOfProc ? resultTask.Result : await resultTask;
```

Comment:
The call to `result.Task.Result` - that's a blocking I/O operation, right? Let's add a comment calling that out, because it could have performance implications and may need to be revisited in the future. IIRC, the reason we needed to do this was because this is an orchestration thread and orchestrators cannot do async I/O (which is why we have the thread ID check that tries to detect this and throw an exception).